test/common/router/rds_impl_test.cc

```diff
@@ -36,9 +41,14 @@ parseHttpConnectionManagerFromJson(const std::string& json_string) {
   return http_connection_manager;
 }
 
-class RdsImplTest : public testing::Test {
+class RdsImplTest : public testing::TestWithParam<Network::Address::IpVersion> {
 public:
-  RdsImplTest() : request_(&cm_.async_client_) {}
+  // TODO(junr03): Switch to mocks and do not bind to a real port.
```

Comment:
I would rewrite these tests to use a mock admin here, not a real one. Then you can undo most/all of these changes. You can verify with mocks adding/removing the handler, and also store the handler to call via test.